earthfile2llb/converter.go

```diff
@@ -1826,30 +1826,30 @@ func (c *Converter) secretID(name string) string {
 }
 
 func (c *Converter) parseSecretFlag(secretKeyValue string) (secretID string, envVar string, err error) {
+	if secretKeyValue == "" {
+		// If empty string, don't use (used for optional secrets).
+		// TODO: This should be an actual secret (with an empty value),
+		//       so that the cache works correctly.
+		return "", "", nil
+	}
 	parts := strings.SplitN(secretKeyValue, "=", 2)
 	if len(parts) == 2 {
-		if strings.HasPrefix(parts[1], "+secrets/") {
-			c.opt.Console.Printf("Deprecation: the '+secrets/' prefix is not required and support for it will be removed in an upcoming release")
-			secretID := strings.TrimPrefix(parts[1], "+secrets/")
-			return secretID, parts[0], nil
-		} else if parts[1] == "" {
+		secretID := parts[1]
+		if secretID == "" {
 			// If empty string, don't use (used for optional secrets).
 			// TODO: This should be an actual secret (with an empty value),
 			//       so that the cache works correctly.
 			return "", "", nil
-		} else {
-			return "", "", errors.Errorf("secret definition %s not supported. Format must be either <env-var>=+secrets/<secret-id> or <secret-id>", secretKeyValue)
 		}
-	} else if len(parts) == 1 {
-		if secretKeyValue == "" {
-			// If empty string, don't use (used for optional secrets).
-			// TODO: This should be an actual secret (with an empty value),
-			//       so that the cache works correctly.
-			return "", "", nil
+		if strings.HasPrefix(secretID, "+secrets/") {
+			c.opt.Console.Printf("Deprecation: the '+secrets/' prefix is not required and support for it will be removed in an upcoming release")
+			secretID = strings.TrimPrefix(secretID, "+secrets/")
 		}
+		return secretID, parts[0], nil
```

Comment:
The fact that `+secrets/` is no longer required is a language change. We should put it behind a feature flag so that it goes out with a new `VERSION` number only. To keep things simple we could put it in the same feature flag that enables the `PROJECT` keyword.

In fact, perhaps the deprecation should only show up when the feature flag is set.